StrongInject.Tests.Unit/OwnedInjectionTests.cs

```diff
@@ -0,0 +1,2042 @@
+";
+            var comp = RunGeneratorWithStrongInjectReference(userSource, out var generatorDiagnostics, out var generated);
+            generatorDiagnostics.Verify();
+            comp.GetDiagnostics().Verify();
+            var file = Assert.Single(generated);
+            file.Should().BeIgnoringLineEndings(@"#pragma warning disable CS1998
+partial class Container
+{
+    private int _disposed = 0;
+    private bool Disposed => _disposed != 0;
+    public void Dispose()
+    {
+        var disposed = global::System.Threading.Interlocked.Exchange(ref this._disposed, 1);
+        if (disposed != 0)
+            return;
+    }
+
+    TResult global::StrongInject.IContainer<global::A>.Run<TResult, TParam>(global::System.Func<global::A, TParam, TResult> func, TParam param)
+    {
+        if (Disposed)
+            throw new global::System.ObjectDisposedException(nameof(Container));
+        global::StrongInject.Owned<global::B> owned_0_1;
+        global::A a_0_0;
+        owned_0_1 = global::Container.CreateOwnedManually();
+        try
+        {
+            a_0_0 = new global::A(B: owned_0_1);
+        }
+        catch
+        {
+            ((global::System.IDisposable)owned_0_1).Dispose();
+            throw;
+        }
+
+        TResult result;
+        try
+        {
+            result = func(a_0_0, param);
+        }
+        finally
+        {
+            ((global::System.IDisposable)owned_0_1).Dispose();
+        }
+
+        return result;
+    }
+
+    global::StrongInject.Owned<global::A> global::StrongInject.IContainer<global::A>.Resolve()
+    {
+        if (Disposed)
+            throw new global::System.ObjectDisposedException(nameof(Container));
+        global::StrongInject.Owned<global::B> owned_0_1;
+        global::A a_0_0;
+        owned_0_1 = global::Container.CreateOwnedManually();
+        try
+        {
+            a_0_0 = new global::A(B: owned_0_1);
+        }
+        catch
+        {
+            ((global::System.IDisposable)owned_0_1).Dispose();
+            throw;
+        }
+
+        return new global::StrongInject.Owned<global::A>(a_0_0, () =>
+        {
+            ((global::System.IDisposable)owned_0_1).Dispose();
+        });
+    }
+}");
+        }
+
+        // Demonstrates another especially useless thing to do as currently producing working code with no StrongInject diagnostics.
+        // (There's no way to provide a semantically correct dispose action if the contained instance or any of its dependencies are disposable.)
+        [Fact]
+        public void OwnedCanBeRegisteredExplicitly()
+        {
+            string userSource = @"
+using System;
+using StrongInject;
+
+[Register(typeof(A)), Register(typeof(B)), Register(typeof(Owned<>))]
+public partial class Container : IContainer<A>
+{
+    [Instance]
+    private static readonly Action NoOp = () => { };
+}
+
+public record A(Owned<B> B);
+public record B : IDisposable { public void Dispose() { } }
+";
```

Comment:
Possible future PR to generate an error diagnostic if `Owned<...>` or `AsyncOwned<...>` are explicitly registered. (https://github.com/YairHalberstadt/stronginject/pull/143#discussion_r700522078)

Reply:
@YairHalberstadt At the moment, I'm not planning on implementing this.

Reply:
Absolutely fine, no expectation that you should.